### PR TITLE
rustfmt: stop pretending we're on nightly

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

Fixes build after #161726.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by K900** [comment]
Still broken: indradb (missing PROTOC binary, probably hasn't built ever?) and bindgen (looks like rustfmt started rewriting doc comments, which broke tests)

File: pkgs/development/compilers/rust/rustfmt.nix [hunk 1]
```diff
@@ -18,8 +18,8 @@ rustPlatform.buildRustPackage rec {
 
   # As of rustc 1.45.0, these env vars are required to build rustfmt (due to
   # https://github.com/rust-lang/rust/pull/72001)
-  CFG_RELEASE = "${rustPlatform.rust.rustc.version}-nightly";
-  CFG_RELEASE_CHANNEL = "nightly";
+  CFG_RELEASE = "${rustPlatform.rust.rustc.version}";
```
Comment (alyssais) [inline pkgs/development/compilers/rust/rustfmt.nix:21]:
```suggestion
  CFG_RELEASE = rustPlatform.rust.rustc.version;
```

**Review by alyssais — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/rust/rustfmt.nix (hunk 1 above)
Comment (K900) [reply pkgs/development/compilers/rust/rustfmt.nix]: Applied.

**Comment by K900** [comment]
Fixed indradb, will poke bindgen tomorrow.

**Review by alyssais — APPROVED** [review]
rustfmt changes look good

**Pull request merged**